flexget/plugins/filter/unique.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+
+from flexget import plugin
+from flexget.event import event
+from flexget.config_schema import one_or_more
+
+log = logging.getLogger('unique')
+
+
+class Unique(object):
+    """
+    Take action on entries with duplicate fields, except for the first item
+
+    Reject the second+ instance of every movie:
+
+      unique:
+        field:
+          - imdb_id
+          - movie_name
+        action: reject
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'field': one_or_more({'type': 'string'}),
+            'action': {'enum': ['accept', 'reject']},
+        },
+        'required': ['field'],
+        'additionalProperties': False
+    }
+
+    def prepare_config(self, config):
+        if isinstance(config, bool) or config is None:
+            config = {}
+        config.setdefault('action', 'reject')
+        if not isinstance(config['field'], list):
+            config['field'] = [config['field']]
+        return config
+
+    def extract_fields(self, entry, field_names):
+        if all ([field in entry for field in field_names]):
+            return [ entry[field] for field in field_names ]
+        else:
+            return None
+
+    def should_ignore(self, item, action):
+        return item.accepted and action == 'accept' or item.rejected and action == 'reject'
+
+    def on_task_filter(self, task, config):
+        config = self.prepare_config(config)
+        field_names = config['field']
+        entries = list(task.entries)
+        for i, entry in enumerate(entries):
+            # Ignore already processed entries
+            if self.should_ignore(entry, config['action']):
+                continue
+            entry_fields = self.extract_fields(entry,field_names)
+            # Ignore if a field is missing
+            if entry_fields is None:
```

Comment:
I think it would be better to use EAFP in this case. Just have extract_fields assume all fields are present and catch the KeyError here.